packages/sdk/src/shared/react/hooks/useCurrencyConversion.ts

```diff
@@ -0,0 +1,101 @@
+import { useExchangeRate } from "@b3dotfun/sdk/global-account/react";
+import { formatNumberWithAbbreviations } from "@b3dotfun/sdk/shared/utils/formatNumberWithAbbreviations";
+import { CURRENCY_SYMBOLS, useCurrencyStore } from "../stores/currencyStore";
+
+export function useCurrencyConversion() {
+  const selectedCurrency = useCurrencyStore(state => state.selectedCurrency);
+  const baseCurrency = useCurrencyStore(state => state.baseCurrency);
+
+  // Use the real exchange rate hook from the SDK
+  // If selected currency is same as base, get USD rate for tooltip conversion
+  const { rate: exchangeRate } = useExchangeRate({
+    baseCurrency,
+    quoteCurrency: selectedCurrency === baseCurrency ? "USD" : selectedCurrency,
+  });
+
+  const formatCurrencyValue = (value: number): string => {
```

Comment:
This function looks complicated without document, can you add document and tests?